sdk/python/kfp/_client.py

```diff
@@ -51,22 +54,23 @@ def __init__(self, host=None, client_id=None, namespace='kubeflow'):
       client_id: The client ID used by Identity-Aware Proxy.
     """
 
-    self._host = host
+    self._uihost = os.environ.get(KF_PIPELINES_UI_ENDPOINT_ENV, host)
     config = self._load_config(host, client_id, namespace)
     api_client = kfp_server_api.api_client.ApiClient(config)
     self._run_api = kfp_server_api.api.run_service_api.RunServiceApi(api_client)
     self._experiment_api = kfp_server_api.api.experiment_service_api.ExperimentServiceApi(api_client)
 
   def _load_config(self, host, client_id, namespace):
     config = kfp_server_api.configuration.Configuration()
+    config.host = config.host or os.environ.get(KF_PIPELINES_ENDPOINT_ENV)
```

Comment:
host = host or os.environ.get(KF_PIPELINES_ENDPOINT_ENV, None)

Otherwise, the code will fail if no host and env var are provided.

Reply:
@hongye-sun i don't think `get` fails, will just return `None`
just try this: `print('' or os.environ.get('SOMEKEY'))` on your laptop

Reply:
you are right on the environ.get. You still need to change the config.host to host or you should change rest of the method body to check against config.host instead of host.

Reply:
@hongye-sun you can see the line below, (`if host: config.host = host`), host will take precedence over the env var, i'm not clear why it doesnt address the case ?

Reply:
Better to go with test cases:

For example, (host=None, os.environ.get(KF_PIPELINES_ENDPOINT_ENV)='https://iap-endpoint/', client_id='<iap client id>'), in ln 70, iap token won't be fetched.

Another example, (host=None, os.environ.get(KF_PIPELINES_ENDPOINT_ENV)='https://iap-endpoint/', client_id=None), in ln 79, it will ignore config.host and override it with either incluster DNS or the default kubu config's  endpoint.

Reply:
@hongye-sun ok, got u, switched to `host`, make more sense